client/src/PublishTenderForm.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+const PublishTenderForm = () => {
+	const [title, setTitle] = useState("");
+	const [description, setDescription] = useState("");
+	const [closingDate, setClosingDate] = useState("");
+	const [announcementDate, setAnnouncementDate] = useState("");
+	const [deadlineDate, setDeadlineDate] = useState("");
+	const [skills, setSkills] = useState([]);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+
+		if (title.length < 10 || title.length > 50) {
+			alert("Tender Title must be between 10 and 50 characters.");
+			return;
+		}
+
+		if (description.length < 100 || description.length > 7500) {
+			alert("Tender Description must be between 100 and 7500 characters.");
+			return;
+		}
+
+		const today = new Date().toISOString().split("T")[0];
+		if (closingDate < today) {
+			alert("Tender Closing Date cannot be in the past.");
+			return;
+		}
+
+		if (announcementDate < closingDate) {
+			alert("Tender Announcement Date cannot be before the Closing Date.");
+			return;
+		}
+
+		if (deadlineDate < announcementDate) {
+			alert(
+				"Tender Project Deadline Date cannot be before the Announcement Date."
+			);
+			return;
+		}
+
+		if (skills.length === 0) {
+			alert("Please select at least one skill.");
+			return;
+		}
+
+		console.log({
+			title,
+			description,
+			closingDate,
+			announcementDate,
+			deadlineDate,
+			skills,
+		});
+
+		alert("Tender published successfully!");
+		
+		setTitle("");
+		setDescription("");
+		setClosingDate("");
+		setAnnouncementDate("");
+		setDeadlineDate("");
+		setSkills([]);
+	};
+
+	const handleSkillsChange = (event) => {
+		const options = event.target.options;
+		const selectedSkills = [];
+		for (const option of options) {
+			if (option.selected) {
+				selectedSkills.push(option.value);
+			}
+		}
+		setSkills(selectedSkills);
+	};
+
+	return (
+		<div className="container">
+			<h1>Publish Tender</h1>
+			<form onSubmit={handleSubmit}>
+				<div className="form-group">
+					<label htmlFor="title">Tender Title:</label>
+					<input
+						type="text"
+						id="title"
+						value={title}
+						onChange={(e) => setTitle(e.target.value)}
```

Comment:
This seems a little wasteful, you are re-rendering on every keystroke. Maybe onBlur would be better

Reply:
I have made the change as requested

Reply:
What you are doing does exactly the same you were doing before. I meant you should use `onBlur` instead of `onChange`

Reply:
We decided to stick with `onChage` for now as it would be a little more work to change this